streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KGroupedStream.scala

```diff
@@ -86,7 +85,7 @@ class KGroupedStream[K, V](val inner: KGroupedStreamJ[K, V]) {
    * @see `org.apache.kafka.streams.kstream.KGroupedStream#reduce`
    */ 
   def reduce(reducer: (V, V) => V,
-    materialized: Materialized[K, V, ByteArrayKeyValueStore]): KTable[K, V] = {
+    materialized: Materialized[K, V, KeyValueStore[Bytes, Array[Byte]]]): KTable[K, V] = {
```

Comment:
I think `ByteArrayKeyValueStore` is more succinct than `KeyValueStore[Bytes, Array[Byte]]`? cc @debasishg @seglo for a second thought.

Reply:
+1 @guozhangwang .. exactly for this reason the type synonym was introduced

Reply:
I'm good with the type alias `ByteArrayKeyValueStore ` but then we should do it consistently for  the other types like `SessionStore[Bytes, Array[Byte]]`. I will then modify the PR to add the others.